### ISSUE TITLE
[BUG] PnP.PowerShell 1.10.0 throws error when connecting from Azure Function 

### ISSUE THREAD

**Issue by michaeltschreiber** [pnp/powershell#2032]
### Reporting an Issue or Missing Feature

PnP.PowerShell 1.10.0 throws error when connecting from Azure Function ERROR: Could not load type 'Microsoft.Extensions.Logging.Abstractions.Internal.NullScope' from assembly 'Microsoft.Extensions.Logging.Abstractions, Version=3.0.3.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'

Runing the same code, I do not encounter this error locally on Windows 10.

EDIT: I have now tested that this error occurs when using the PnP.PowerShell module in the context of a thread job per the PnP.PowerShell documentation, and the error persists.

### Expected behavior 
Connect-PnPOnline cmdlet successfully establishes SharePoint Connection

### Actual behavior 

Error is thrown as shown here:
![image](https://user-images.githubusercontent.com/68429044/175571668-e6e21e3a-7294-43d1-b04f-173bbc553dcc.png)

### Steps to reproduce behavior

![image](https://user-images.githubusercontent.com/68429044/175571090-6510d819-d79c-426d-b30e-8132c2d13a43.png)

### What is the version of the Cmdlet module you are running?
1.10.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [X ] Azure Functions
- [ ] Other : please specify

**Comment by KoenZomers** [comment]
Interesting. I'm using this all the time myself without any issues. Do you have your Azure Function configured to use PowerShell 7?

**Comment by KoenZomers** [comment]
You might also want to check your requirements.psd1 file that you do not have any other PowerShell modules included in it:

<img width="435" alt="image" src="https://user-images.githubusercontent.com/5940670/175804945-24187abe-92e7-4f2d-91e5-da87082848c4.png">

**Comment by michaeltschreiber** [comment]
> You might also want to check your requirements.psd1 file that you do not have any other PowerShell modules included in it:
> 
> <img alt="image" width="435" src="https://user-images.githubusercontent.com/5940670/175804945-24187abe-92e7-4f2d-91e5-da87082848c4.png">

Thank you so much for your reply! I actually was not using managed dependencies, but based on your suggestion I removed my modules folder and added the managed dependency as you suggested.

Unfortunately, I receive the same error:
![image](https://user-images.githubusercontent.com/68429044/175822438-217cd99f-2e94-44ab-a53f-db5dd863ddf8.png)

**Comment by michaeltschreiber** [comment]
I have now resolved my own issue.

It turns out that when I initially configured this function I inadvertently selected. PowerShell 7.2 as the PowerShell Core Version. Editing my configuration at Configuration > General Settings > PowerShell Core Version and setting the value "PowerShell 7.0" resolved my issue, and I was able to execute my function without any type load error.
![image](https://user-images.githubusercontent.com/68429044/175822900-b17a4680-a871-423c-a5a4-6eaf4b8fc7ae.png)

![image](https://user-images.githubusercontent.com/68429044/175822978-de3d5c5d-9741-45e1-9c9a-924eabf72e6a.png)



Be advised there is currently an incompatibility between latest version of the PnP.PowerShell framework and the latest version of PowerShell Core available in the Azure Functions runtime.

Interestingly, I do not encounter this bug running locally on PowerShell 7.2.4.
